File: fix_all_issues.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_file_issues(file_path):
    """Fix common issues in React/TypeScript files"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix duplicate default exports
        if content.count('export default') > 1:
            # Keep only the first export default
            lines = content.split('\n')
            export_default_count = 0
            new_lines = []
            
            for line in lines:
                if line.strip().startswith('export default') and export_default_count > 0:
                    # Skip duplicate export default lines
                    continue
                elif line.strip().startswith('export default'):
                    export_default_count += 1
                    new_lines.append(line)
                else:
                    new_lines.append(line)
            
            content = '\n'.join(new_lines)
        
        # Fix missing imports for common components
        if 'Helmet' in content and 'react-helmet-async' not in content:
            if 'import { Helmet }' not in content:
                # Add Helmet import
                import_line = "import { Helmet } from 'react-helmet-async'"
                if 'import React' in content:
                    content = content.replace('import React', f'import React\n{import_line}')
                else:
                    content = f"{import_line}\n{content}"
        
        if 'Link' in content and 'react-router-dom' not in content:
            if 'import { Link }' not in content:
                # Add Link import
                import_line = "import { Link } from 'react-router-dom'"
                if 'import React' in content:
                    content = content.replace('import React', f'import React\n{import_line}')
                else:
                    content = f"{import_line}\n{content}"
        
        # Remove any remaining merge conflict markers

        # Clean up multiple empty lines
        content = re.sub(r'\n\n\n+', '\n\n', content)
        
        # Remove trailing whitespace
        content = re.sub(r'[ \t]+$', '', content, flags=re.MULTILINE)
        
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed issues in: {file_path}")
            return True
        
        return False
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    # Find all TypeScript/JavaScript files
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts',
        'app/**/*.js',
        'app/**/*.jsx'
    ]
    
    fixed_count = 0
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if fix_file_issues(file_path):
                fixed_count += 1
    
    print(f"Fixed issues in {fixed_count} files")

if __name__ == "__main__":
    main()

File: fix-all-issues.js
import fs from 'fs';
import path from 'path';
import { execSync } from 'child_process';

// Find all page.tsx files
function findPageFiles(dir) {
  const files = [];
  const items = fs.readdirSync(dir);
  
  for (const item of items) {
    const fullPath = path.join(dir, item);
    const stat = fs.statSync(fullPath);
    
    if (stat.isDirectory()) {
      files.push(...findPageFiles(fullPath));
    } else if (item === 'page.tsx') {
      files.push(fullPath);
    }
  }
  
  return files;
}

// Fix all issues in a file
function fixAllIssues(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;
    
    // Fix ErrorBoundary import path
    content = content.replace(/import { ErrorBoundary } from '@\/components\/ErrorBoundary';/g, "import ErrorBoundary from '@/components/ErrorBoundary';");
    
    // Remove duplicate metadata declarations (keep only the first one)
    const metadataRegex = /export const metadata = \{[\s\S]*?\};/g;
    const matches = content.match(metadataRegex);
    if (matches && matches.length > 1) {
      // Keep only the first metadata declaration
      const firstMetadata = matches[0];
      content = content.replace(metadataRegex, '');
      // Add the first metadata back at the top after imports
      const importEnd = content.indexOf(';', content.indexOf('import')) + 1;
      content = content.slice(0, importEnd) + '\n\n' + firstMetadata + '\n' + content.slice(importEnd);
    }
    
    // Fix Page component reference
    content = content.replace(/<Page \{\.\.\.props\} \/>/g, '<PageComponent {...props} />');
    
    // Fix function name if it's just 'Page'
    content = content.replace(/function Page\(\)/g, 'function PageComponent()');
    
    // Fix unused variable names
    content = content.replace(/const pagePage =/g, 'const pageComponent =');
    content = content.replace(/pagePage/g, 'pageComponent');
    
    // Clean up extra semicolons and empty lines
    content = content.replace(/;\s*;/g, ';');
    content = content.replace(/\n\s*\n\s*\n/g, '\n\n');
    
    // If we made changes, write the file back
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Main function
function main() {
  console.log('Starting to fix all issues...');
  
  const appDir = path.join(process.cwd(), 'app');
  const pageFiles = findPageFiles(appDir);
  
  console.log(`Found ${pageFiles.length} page files`);
  
  let fixedCount = 0;
  
  for (const filePath of pageFiles) {
    if (fixAllIssues(filePath)) {
      fixedCount++;
    }
  }
  
  console.log(`Fixed ${fixedCount} files`);
  
  // Run type check to verify
  try {
    console.log('Running type check...');
    execSync('npm run type-check', { stdio: 'inherit' });
    console.log('Type check passed!');
  } catch (error) {
    console.error('Type check failed:', error.message);
  }
}

main();